### PR TITLE
Create Issue type when saving

### PR DESCRIPTION
**Pull request by Daverlo**
@aibaars

### CODE REVIEW

File: src/main/java/com/semmle/jira/addon/util/JiraUtils.java [hunk 1]
```diff
@@ -43,16 +45,25 @@ public static IssueType getIssueTypeByName(String issueTypeName) {
...
     GenericValue workflowScheme = workflowSchemeManager.getWorkflowScheme(project);
-    workflowSchemeManager.addWorkflowToScheme(
-        workflowScheme, workflow.getName(), issueType.getId());
-  }
-
-  public static IssueType getLgtmIssueType() {
-    return getIssueTypeByName(Constants.ISSUE_TYPE_NAME);
+    Map<String, String> workflowMap = workflowSchemeManager.getWorkflowMap(project);
+    if (workflowMap.containsKey(issueType.getId())) {
+      if (!workflowMap.get(issueType.getId()).equals(workflow.getName())) {
```
Comment (nickfyson) [inline src/main/java/com/semmle/jira/addon/util/JiraUtils.java]: I'm not sure this achieves quite what we want (unless I'm missing something). If the `IssueType` has not been explicitly associated with a workflow scheme, perhaps it won't appear in the `workflowMap`, but nonetheless falls back to the `default` scheme?

Either way, trying this with existing tickets in a local Jira instance, I didn't get the required error when saving a new config.

Comment (nickfyson) [reply src/main/java/com/semmle/jira/addon/util/JiraUtils.java]: I thought if there are any existing tickets of type `LGTM alert` then we would need to raise this for manual intervention by the user?

Comment (Daverlo) [reply src/main/java/com/semmle/jira/addon/util/JiraUtils.java]: It only throws an error when the `LGTM alert` issue type is associated with a workflow that is not `LGTM alert`.

Comment (Daverlo) [reply src/main/java/com/semmle/jira/addon/util/JiraUtils.java]: What error are you missing?

Comment (nickfyson) [reply src/main/java/com/semmle/jira/addon/util/JiraUtils.java]: I was expecting an error on trying to save plugin config when I already had tickets of type `LGTM alert`.

Perhaps I'm wrong in expecting that to throw an error? What guarantees do we have about the workflow of existing tickets if this check passes?

Comment (aibaars) [reply src/main/java/com/semmle/jira/addon/util/JiraUtils.java]: I think the `if` branch could be removed now, as the `else` branch should cover the dangerous case where issues already exist.

File: src/main/java/com/semmle/jira/addon/util/JiraUtils.java [hunk 2]
```diff
@@ -43,16 +51,41 @@ public static IssueType getIssueTypeByName(String issueTypeName) {
...
+        throw new UsedIssueTypeException();
+      }
+    } else {
+      JqlQueryBuilder builder = JqlQueryBuilder.newBuilder();
+      builder.where().project(project.getId()).and().issueType(issueType.getName());
+      try {
+        SearchResults results =
+            ComponentAccessor.getComponent(SearchService.class)
+                .searchOverrideSecurity(
+                    user, builder.buildQuery(), PagerFilter.getUnlimitedFilter());
```
Comment (aibaars) [inline src/main/java/com/semmle/jira/addon/util/JiraUtils.java]: A `limited` pagerfilter might be better, we only need to know if any result exists.

**Pull request merged**